mne/decoding/receptive_field.py

```diff
@@ -0,0 +1,363 @@
+            raise ValueError('X must be shape (n_times, [n_epochs],'
+                             ' n_features)')
+        elif X.ndim == 2:
+            # Ensure we have a 3D input by adding singleton epochs dimension
+            X = X[:, np.newaxis, :]
+            if y is None:
+                pass
+            elif y.ndim == 1:
+                y = y[:, np.newaxis, np.newaxis]  # Add epochs and outputs dim
+            elif y.ndim == 2:
+                y = y[:, np.newaxis]  # Only add epochs dim
+            else:
+                raise ValueError('y must be shape (n_times, [n_epochs],'
+                                 ' n_outputs')
+        elif X.ndim == 3:
+            if y is None:
+                pass
+            elif y.ndim == 2:
+                y = y[:, :, np.newaxis]  # Add an outputs dim
+            elif y.ndim != 3:
+                raise ValueError('If X has 3 dimensions, '
+                                 'y must be at least 2 dimensions')
+        else:
+            raise ValueError('X must be of shape '
+                             '(n_times, [n_epochs], n_features)')
+        if y is None:
+            # If y is None, then we don't need any more checks
+            return X, y
+
+        if X.shape[0] != y.shape[0]:
+            raise ValueError('X any y do not have the same n_times\n'
+                             '%s != %s' % (X.shape[0], y.shape[0]))
+        if X.shape[1] != y.shape[1]:
+            raise ValueError('X any y do not have the same n_epochs\n'
+                             '%s != %s' % (X.shape[1], y.shape[1]))
+        if predict is True:
+            if y.shape[-1] != len(self.estimator_.coef_):
+                raise ValueError('Number of outputs does not match'
+                                 ' estimator coefficients dimensions')
+        return X, y
+
+
+def _delay_time_series(X, tmin, tmax, sfreq, newaxis=0, axis=0):
+    """Return a time-lagged input time series.
+
+    Parameters
+    ----------
+    X : array, shape (n_times, [n_epochs], n_features)
+        The time series to delay.
+    tmin : int | float
+        The starting lag. Negative values correspond to times in the past.
+    tmax : int | float
+        The ending lag. Positive values correspond to times in the future.
+        Must be >= tmin.
+    sfreq : int | float
+        The sampling frequency of the series. Defaults to 1.0.
+    newaxis : int
+        The axis in the output array that corresponds to time delays.
+        Defaults to 0, for the first axis.
+    axis : int
+        The axis corresponding to the time dimension.
+
+    Returns
+    -------
+    delayed: array, shape(..., n_delays, ...)
+        The delayed data. It has the same shape as X, with an extra dimension
+        created at ``newaxis`` that corresponds to each delay.
+    """
+    _check_delayer_params(tmin, tmax, sfreq)
+    delays = _times_to_delays(tmin, tmax, sfreq)
+    # XXX : add Vectorize=True parameter to switch on/off 2D output
```

Comment:
yeah, this is in case you want to force the output to be 2d for including in pipelines etc (I think). We aren't using it for any of the examples / tutorials with this contribution but I could imagine it being useful in the future.